File: maven-leetcode/src/main/java/leetcode/Num35.java
package leetcode;

public class Num35 {
    class Solution {
        public int searchInsert(int[] nums, int target) {
            int high = nums.length -1;
            int low = 0;
            int mid = -1;
            while(low <= high){
                mid = (low + high) / 2;
                if (nums[mid] > target) {
                    high = mid - 1;
                } else if (nums[mid] < target) {
                    low = mid + 1;
                } else {
                    break;
                }
            }
            if (low <= high) {
                return mid;
            } else {
                return low;
            }

        }
    }
}

File: learnPython/num35.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        left = 0
        right = len(nums)-1
        while left <= right:
            medium = (left+right)//2
            if nums[medium] == target:
                return medium
            elif nums[medium] < target:
                left = medium + 1
            else:
                right = medium - 1
        return left
if __name__ == '__main__':
    l = [1,3,5,6]
    s = Solution()
    print(s.searchInsert(l,2))